stl/inc/xutility

```diff
@@ -5220,15 +5220,28 @@ namespace ranges {
     template <input_iterator _It, sentinel_for<_It> _Se, class _Ty, class _Pj = identity>
         requires indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>
     _NODISCARD constexpr _It _Find_unchecked(_It _First, const _Se _Last, const _Ty& _Val, _Pj _Proj = {}) {
-        if constexpr (_Memchr_in_find_is_safe<_It, _Ty> && sized_sentinel_for<_Se, _It> && same_as<_Pj, identity>) {
+        constexpr bool _Is_sized = sized_sentinel_for<_Se, _It>;
+        if constexpr (_Memchr_in_find_is_safe<_It, _Ty> && (_Is_sized || same_as<_Se, unreachable_sentinel_t>)
+                      && same_as<_Pj, identity>) {
             if (!_STD is_constant_evaluated()) {
                 if (!_Within_limits(_First, _Val)) {
-                    return _RANGES next(_STD move(_First), _Last);
+                    if constexpr (_Is_sized) {
+                        return _RANGES next(_STD move(_First), _Last);
+                    } else {
+                        _STL_ASSERT(false, "Tried to find a value in a range with unreachable sentinel"

```

Comment:
This should be a static assert. It is incorrect regardless of `STD is_constant_evaluated()`

Reply:
Ditto